src/pt/sussyscan/src/eu/kanade/tachiyomi/extension/pt/sussyscan/SussyScan.kt

```diff
@@ -1,34 +1,207 @@
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    // ============================= Search ===================================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/obras".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "8")
+            .addQueryParameter("obr_nome", query)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+
+    // ============================= Details ==================================
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+        .replace("obras", "obra")
+
+    override fun mangaDetailsRequest(manga: SManga) =
+        GET("$apiUrl${manga.url.substringBeforeLast("/")}", headers)
+
+    override fun mangaDetailsParse(response: Response) =
+        toSManga(response.parseAs<WrapperDto<MangaDto>>().results)
 
-    override val mangaDetailsSelectorTitle = "${super.mangaDetailsSelectorTitle}, span.rate-title, title"
-    override val mangaDetailsSelectorThumbnail = "head meta[property='og:image']"
+    // ============================= Chapters =================================
 
-    override fun mangaDetailsParse(document: Document) = super.mangaDetailsParse(document).apply {
-        title = title.substringBeforeLast("–")
+    override fun getChapterUrl(chapter: SChapter) = "$baseUrl${chapter.url}"
+        .replace("capitulos", "capitulo")
+
+    override fun chapterListRequest(manga: SManga) = throw UnsupportedOperationException()
+
+    private fun chapterListRequest(page: Int, manga: SManga): Request {
+        val scanId = manga.url.substringAfter("#")
+        val url = "$apiUrl/obras/${manga.getId()}/capitulos".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "100")
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return response.parseAs<WrapperDto<List<ChapterDto>>>().results.map {
+            SChapter.create().apply {
+                name = it.name
+                it.chapterNumber?.let {
+                    chapter_number = it
+                }
+                url = "/capitulos/${it.id}"
+                date_upload = it.updateAt.toDate()
+            }
+        }
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapterList = mutableListOf<SChapter>()
+        var page = 1
+        do {
+            val chapters = client.newCall(chapterListRequest(page++, manga))
+                .execute().let(::chapterListParse)
+            chapterList += chapters
+        } while (chapters.isNotEmpty())
+
+        return Observable.just(chapterList.reversed())
```

Comment:
Will this be in the right order? Just make sure you aren't reversing it page by page, but rather as a whole.